src/tasty/utils/cache-wrapper.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Create a function that caches the result up to the limit.
+ */
+export function cacheWrapper<
+  T extends (firstArg: any, secondArg?: string) => any,
+>(handler: T, limit = 1000): T {
+  let cache: { string?: ReturnType<T> } = {};
+  let count = 0;
+
+  return ((firstArg: any, secondArg?: string) => {
+    const key =
+      typeof firstArg === 'string' && secondArg == null
+        ? firstArg
+        : JSON.stringify([firstArg, secondArg]);
+
+    if (!cache[key]) {
```

Comment:
I think we should consider smth like a lru cache here, wdyt?

Reply:
It's the old code. It was moved here.
We can add LRU cache later if you know how to make it without affecting performance.

Reply:
>if you know how to make it without affecting performance.

WDYM? `lru-cache` costs O(1) for read/insert in worst case